nodejs/awsx/autoscaling/autoscaling.ts

```diff
@@ -16,63 +16,77 @@ import * as aws from "@pulumi/aws";
 import * as pulumi from "@pulumi/pulumi";
 
 import * as x from "..";
-import * as utils from "./../utils";
+import * as utils from "../utils";
 
 import { AutoScalingLaunchConfiguration, AutoScalingLaunchConfigurationArgs } from "./launchConfiguration";
 import { cronExpression, ScheduleArgs } from "./schedule";
 import * as stepScaling from "./stepScaling";
 import * as targetTracking from "./targetTracking";
 
 export class AutoScalingGroup extends pulumi.ComponentResource {
-    public readonly vpc: x.ec2.Vpc;
+    public readonly vpc!: x.ec2.Vpc;
 
     /**
      * The [cloudformation.Stack] that was used to create this [AutoScalingGroup].  [CloudFormation]
      * is used here as the existing AWS apis for creating [AutoScalingGroup]s are not rich enough to
      * express everything that can be configured through [CloudFormation] itself.
      */
-    public readonly stack: aws.cloudformation.Stack;
+    public readonly stack!: aws.cloudformation.Stack;
 
     /**
      * The launch configuration for this auto scaling group.
      */
-    public readonly launchConfiguration: AutoScalingLaunchConfiguration;
+    public readonly launchConfiguration!: AutoScalingLaunchConfiguration;
 
     /**
      * Underlying [autoscaling.Group] that is created by cloudformation.
      */
-    public readonly group: aws.autoscaling.Group;
+    public readonly group!: aws.autoscaling.Group;
 
     /**
      * Target groups this [AutoScalingGroup] is attached to.  See
      * https://docs.aws.amazon.com/autoscaling/ec2/userguide/attach-load-balancer-asg.html
      * for more details.
      */
-    public readonly targetGroups: x.lb.TargetGroup[];
+    public readonly targetGroups!: x.lb.TargetGroup[];
 
-    constructor(name: string,
-                args: AutoScalingGroupArgs,
-                opts: pulumi.ComponentResourceOptions = {}) {
+    /** @internal */
+    constructor(version: number, name: string, opts: pulumi.ComponentResourceOptions) {
         super("awsx:x:autoscaling:AutoScalingGroup", name, {}, opts);
 
-        this.vpc = args.vpc || x.ec2.Vpc.getDefault({ parent: this });
+        if (typeof version !== "number") {
+            throw new pulumi.ResourceError("Do not call [new AutoScalingGroup] directly. Use [AutoScalingGroup.create] instead.", this);
+        }
```

Comment:
All components have a non-exported constructor, and take a new `version: number` arg as their first parameter.  If this is missing an exception is throw which tells the user to use the factory method instead.